core/src/main/java/com/onelogin/saml2/authn/SubjectConfirmationIssue.java

```diff
@@ -0,0 +1,34 @@
+package com.onelogin.saml2.authn;
+
+import java.util.List;
+
+class SubjectConfirmationIssue {
+    private final int subjectConfirmationIndex;
+    private final String message;
+
+    SubjectConfirmationIssue(int subjectConfirmationIndex, String message) {
+        this.subjectConfirmationIndex = subjectConfirmationIndex;
+        this.message = message;
+    }
+
+    static String prettyPrint(List<SubjectConfirmationIssue> subjectConfirmationDataIssues) {
+        StringBuilder subjectConfirmationDataIssuesMsg = new StringBuilder("A valid SubjectConfirmation was not found on this Response");
+        if (subjectConfirmationDataIssues.size() > 0) {
+            subjectConfirmationDataIssuesMsg.append(" - ");
```

Comment:
Maybe instead of adding " - ", we can add ":" and a end of line, so we will have this kind of output:
```  
A valid SubjectConfirmation was not found on this Response:
        SubjectConfirmationData doesn't contain a NotOnOrAfter attribute [0]
        SubjectConfirmationData doesn't contain a Recipient [1]
        SubjectConfirmationData is no longer valid [2]
```

And maybe index should appear at the begining of the line...